onnxruntime/test/platform/env_test.cc

```diff
@@ -33,5 +33,13 @@ TEST(PlatformEnvTest, DirectoryCreationAndDeletion) {
   ASSERT_FALSE(env.FolderExists(root_dir));
 }
 
+TEST(PlatformEnvTest, GetErrnoInfo) {
+  // command that should generate an errno error
+  std::ifstream file("non_existent_file");
+  ASSERT_TRUE(file.fail());
+  auto [err, msg] = GetErrnoInfo();
+  ASSERT_NE(err, 0);
+  ASSERT_EQ(msg, "No such file or directory");
```

Comment:
the error message is locale specific. maybe check for err == ENOENT instead?

Reply:
I wanted to check that a string was set and was valid. Is there an `if` condition that could limit that to a specific locale?

Reply:
hm, I think setlocale can query it. not sure what value we should test for though.
https://en.cppreference.com/w/cpp/locale/setlocale

maybe another approach is to explicitly set it to be a certain locale for this test, and then reset it to whatever it was before? e.g., the "C" locale.

Reply:
also, I think the error message is implementation-defined